### DISCUSSION TITLE
need more granularity on `error: undefined` when running `npm init directus-project`

### DISCUSSION THREAD

**Discussion by Cerchie** [directus/directus#16588] [Archive]
Hey folks -- I think some more granularity on this error may be helpful. 

I'm currently on an MB Pro Mojave 10.14.6. I figure this error is user error. I couldn't tell from glancing at the documentation but I don't have a cloud account so maybe that's it? Either way, a more detailed error would be helpful for the user. If agreed, I'm happy to write this up into an issue. 

```
Lucias-MacBook-Pro:~ luciacerchie$ npm init directus-project my-directus-project
dyld: lazy symbol binding failed: Symbol not found: _pthread_jit_write_protect_supported_np
  Referenced from: /Users/luciacerchie/my-directus-project/node_modules/sharp/build/Release/../.././vendor/8.13.3/darwin-x64/lib/libvips-cpp.42.dylib
  Expected in: /usr/lib/libSystem.B.dylib

dyld: Symbol not found: _pthread_jit_write_protect_supported_np
  Referenced from: /Users/luciacerchie/my-directus-project/node_modules/sharp/build/Release/../.././vendor/8.13.3/darwin-x64/lib/libvips-cpp.42.dylib
  Expected in: /usr/lib/libSystem.B.dylib

✖ Installing Directus
Error: undefined

```

**Comment by rijkvanzanten** [comment]
This seems to be an issue in installing one of the dependencies of Directus itself (sharp in this case). Can you try `npm install directus && directus init` instead?

**Reply by Cerchie** [reply]
Thanks for the swift response @rijkvanzanten !

That command outputs:

```
$ npm install directus && directus init

up to date, audited 1017 packages in 7s

134 packages are looking for funding
  run `npm fund` for details

2 moderate severity vulnerabilities

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
-bash: directus: command not found
```

On the other hand, I started this discussion to see if it would be worth it to throw a more detailed error-- should I write an issue for that?

**Reply by paescuj** [reply]
The command supposed to be `npm install directus && npx directus init`. Can you try it again with this?

**Reply by Cerchie** [reply]
Thank you @paescuj ! Here's the updated output. 

```
npm install directus && npx directus init

npm WARN deprecated @types/cssnano@5.1.0: This is a stub types definition. cssnano provides its own type definitions, so you do not need this installed.
npm WARN deprecated @npmcli/move-file@1.1.2: This functionality has been moved to @npmcli/fs
npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated keyv-memcache@1.2.5: This project has been renamed to @keyv/memcache which will be updated moving forward.
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142

added 1001 packages, and audited 1017 packages in 48s

134 packages are looking for funding
  run `npm fund` for details

2 moderate severity vulnerabilities

To address all issues, run:
  npm audit fix

Run `npm audit` for details.
dyld: lazy symbol binding failed: Symbol not found: _pthread_jit_write_protect_supported_np
  Referenced from: /Users/luciacerchie/node_modules/sharp/build/Release/../.././vendor/8.13.3/darwin-x64/lib/libvips-cpp.42.dylib
  Expected in: /usr/lib/libSystem.B.dylib

dyld: Symbol not found: _pthread_jit_write_protect_supported_np
  Referenced from: /Users/luciacerchie/node_modules/sharp/build/Release/../.././vendor/8.13.3/darwin-x64/lib/libvips-cpp.42.dylib
  Expected in: /usr/lib/libSystem.B.dylib

Abort trap: 6
```

**Reply by paescuj** [reply]
Thanks! It looks like there are some issues with your Node.js / npm installation. How did you install Node.js and which versions are you using (you can run `node -v && npm -v` to get this info)?

**Reply by Cerchie** [reply]
of course, my apologies for not including those in the first place
```
$ node -v && npm -v
v16.14.0
8.3.1
```

**Reply by paescuj** [reply]
Thanks! Looks good so far. And how did you install Node.js (for example via nvm or hombrew)? It might be worth to reinstall Node.js as it seems like there's something wrong with the installation.

**Reply by borisNG** [reply]
i'm having the same issue while trying to install directus locally. do you have any clue on what could be the cause of the issue ?

**Reply by paescuj** [reply]
> i'm having the same issue while trying to install directus locally. do you have any clue on what could be the cause of the issue ?

How did you install Node.js (for example via nvm or hombrew)? Have you already tried to reinstall Node.js? This error message indicates that something might be wrong with the installation...

**Reply by borisNG** [reply]
i've installed nodeJS via nvm. i've used many differents versions of nodeJS from 14 to 18(via nvm) but none of those work.

**Reply by paescuj** [reply]
As already indicated above, this means that there's something wrong with the Node.js installation. Googling this error message leads to recommendations such as ensuring `xcode` is installed properly (see https://github.com/fsevents/fsevents/issues/313#issuecomment-609452221) or reinstalling Node.js (see https://stackoverflow.com/questions/30130692/dyld-lazy-symbol-binding-failed-symbol-not-found-node-module-register).

**Note:** This issue is not related to Directus.

**Comment by paescuj** [comment]
> [...] Either way, a more detailed error would be helpful for the user. If agreed, I'm happy to write this up into an issue.

Regarding this concern: I agree that `Error: undefined` is not very meaningful, so I've submitted [a pull request](https://github.com/directus/create-directus-project/pull/7) in order to improve the error messages.
